Automattic/wp-calypso
# Run headstart after `design-setup-site` step

#### Changes proposed in this Pull Request

* Split `setDesignOnSite()` out of the `setThemeOnSite()` function, based on #56160 
* After completing the `design-setup-site` step we apply the theme and then run headstart
* Use `dont_change_homepage` flag when switching theme since there's no point changing the homepage if we're about to run headstart

Running headstart after selecting a design means all the expected pages are set on the site and the old homepage isn't left over as a draft.

When this PR is paired with #56233 it means that the design picker will only show themes, and selecting a theme will make the site appear as it would have if the site had been initially created with that theme.

#### Testing instructions



* `/start/setup-site?siteSlug=<< site slug >>`
* Choose design
* After landing in My Home check that
	* the theme has been set correctly
	* there are no draft pages floating around
	* all pages and posts that appear on the theme's demo site appear on the newly setup site
